### PR TITLE
feat(web_console): federation filter for tickets

### PR DESCRIPTION
**Pull request by AlllenShen**
* add global store

### CODE REVIEW

File: web_console/pages/job/index.jsx [hunk 1]
```diff
@@ -76,12 +76,12 @@ function useStyles(theme) {
...
 }
 
-function JobList(props) {
+export default function JobList(props) {
   const theme = useTheme();
   const styles = useStyles(theme);
 
   const { data, mutate } = useSWR('jobs', fetcher);
-  const jobs = data ? data.data.filter(x => x.metadata) : null;
+  const jobs = data ? data.data : null
```
Comment (marswong) [inline web_console/pages/job/index.jsx:84]: keep the filter, or the page would crash. the whole page is heavily rely on k8s metadata currently

File: web_console/pages/raw_data/index.jsx [hunk 2]
```diff
@@ -46,14 +48,16 @@ export default function RawDataList() {
...
-    ? rawDatas.map((x) => {
-      const context = JSON.stringify(x.context);
-      return {
-        ...x,
-        context,
-        operation,
-      };
-    })
+    ? rawDatas
+      .filter(el => el.federation_id === federationID)
```
Comment (marswong) [inline web_console/pages/raw_data/index.jsx:52]: no need to filter at page

File: web_console/components/TicketList.jsx [hunk 3]
```diff
@@ -5,8 +5,10 @@ import Form from './Form';
 import { fetcher } from '../libs/http';
```
Comment (marswong) [inline web_console/components/TicketList.jsx:5]: this file is deprecated now

File: web_console/components/Header.jsx [hunk 4]
```diff
@@ -152,6 +153,22 @@ export default function Header() {
...
+  const displayFederationFilter = [
+    '/job',
+    '/ticket',
+    '/raw_data',
+  ].some(el => el === route)
+  const { data: fedData } = useSWR('federations', fetcher)
+  const federations = fedData ? fedData.data : null
+  const [currFederation, setCurrFederation] = useState(-1)
+  useEffect(() => {
+    setTimeout(() => setCurrFederation(parseInt(localStorage.getItem('federationID')) || -1))
```
Comment (marswong) [inline web_console/components/Header.jsx:165]: why use `setTimeout` inside `useEffect`?

File: web_console/libs/http.js [hunk 5]
```diff
@@ -5,4 +5,10 @@ export const client = ky.create({
   throwHttpErrors: false,
 });
 
-export const fetcher = (url, options = {}) => client.get(url, options).json();
+export const fetcher = (url, options = {}) => {
+  options.headers = {
+    ...options.headers,
+    'X-Federation-Id': localStorage.getItem('federationID')
```
Comment (marswong) [inline web_console/libs/http.js:11]: header is string value, so null value would be converted to `"null"`, `X-Federation-Id` could only be number. ways to fix this:
- don't set `X-Federation-Id` header for non-numeric value (**recommended**)
- convert non-numeric value to `""`

**Review by marswong — CHANGES_REQUESTED** [review]

File: web_console/components/Header.jsx (hunk 4 above)
Comment (AlllenShen) [reply web_console/components/Header.jsx:165]: cause we use ssr, localStorage is not available while rendering. Here needs to get storage after mount and fire rerender.

File: web_console/pages/job/index.jsx (hunk 1 above)
Comment (AlllenShen) [reply web_console/pages/job/index.jsx:84]: a mistake here

File: web_console/components/TicketList.jsx (hunk 3 above)
Comment (AlllenShen) [reply web_console/components/TicketList.jsx:5]: what means? this import is work here

File: web_console/components/Header.jsx (hunk 4 above)
Comment (marswong) [reply web_console/components/Header.jsx:165]: what's the problem? useEffect's callback would only be executed after component mounted

Comment (AlllenShen) [reply web_console/components/Header.jsx:165]: i didn't use `setTimeout` beginning, but `setCurrFederation` did not fire the rerender

Comment (AlllenShen) [reply web_console/components/Header.jsx:165]: i tried again, it works as expected. i'll modify here

File: web_console/pages/raw_data/index.jsx [hunk 6]
```diff
@@ -8,6 +8,7 @@ import Form from '../../components/Form';
 import { fetcher } from '../../libs/http';
 import Empty from '../../components/Empty';
 import { createRawData } from '../../services/raw_data';
+import { useStateValue, StateProvider } from '../store'
```
Comment (marswong) [inline web_console/pages/raw_data/index.jsx:11]: no use for now

**Review by marswong — APPROVED** [review]
LGTM

**Pull request merged**